rtl/__init__.py

```diff
@@ -279,37 +280,66 @@ def vhdlentityfiles(self,value):
-    #            self.print_log(type='O',msg='Loading the deprecated file for now.')
-    #            self._interactive_control_file = obsoletepath
-    #        else:
-    #            self._interactive_control_file = '%s/dofile.do' % self.simpath
-    #    return self._interactive_control_file
-    #@interactive_control_file.setter
-    #def interactive_control_file(self,value): 
-    #    self._interactive_control_file = value
+    @property
+    def interactive_control_contents(self):
+        ''' Content of the interactive rtl control file (.do -file).
+
+        If this property is set, a new dofile gets written to the simulation
+        path. This takes precedence over the file pointed by
+        `interactive_controlfile`.
+
+        For example, the contents can be defined in the top testbench as::
+
+            self.interactive_control_contents="""
+                add wave -position insertpoint \\
+                sim/:tb_inverter:A \\
+                sim/:tb_inverter:clock \\
+                sim/:tb_inverter:Z
+                run -all
+                wave zoom full
+            """
+        
+        '''
+        if not hasattr(self, '_interactive_control_contents'):
+            self._interactive_control_contents = ''
+        return self._interactive_control_contents
+    @interactive_control_contents.setter
+    def interactive_control_contents(self,value): 
+        self._interactive_control_contents = value
+
+    @property
+    def interactive_controlfile(self):
+        ''' Path to interactive rtl control file (.do -file).
+
+        The content of the file can be defined in `interactive_control`. If the
+        content is not set in `interactive_control` -property, the do-file is
+        read from this file path. Default path is `./dofile.do`.
+        '''
+        if not hasattr(self, '_interactive_controlfile'):
+            obsoletepath = '%s/Simulations/rtlsim/dofile.do' % self.entitypath
+            dofilepath = '%s/dofile.do' % self.entitypath
+            if os.path.exists(obsoletepath):
+                self.print_log(type='O',msg='Found obsoleted do-file in %s' % obsoletepath)
+                self.print_log(type='O',msg='To fix the obsolete warning:')
+                self.print_log(type='O',msg='Move the obsoleted file %s to the default path %s' % (obsoletepath,dofilepath))
+                self.print_log(type='O',msg='Or, set a custom do-file path to self.interactive_controlfile.')
+                self.print_log(type='O',msg='Or, define the do-file contents in self.interactive_control_contents in your testbench.')
+                self.print_log(type='O',msg='Using the obsoleted file for now.')
+                dofilepath = obsoletepath
+            elif self.interactive_control_contents != '':
+                if os.path.isfile(dofilepath):
+                    self.print_log(type='W',msg='Interactive control file %s ignored and interactive_control_contents used instead.' % dofilepath)
```

Comment:
This warning is not printed if `interactive_controlfile` is provided as well. Although, in that case the current code seems to silently default to using the dofile pointed by `interactive_controlfile` instead.

Reply:
It should be fixed now